app/controllers/profile.js

```diff
@@ -55,6 +60,53 @@ export default Controller.extend({
     });
   },
 
+  setEmailPendingStatus: task(function * (email) {
+    let url = `${config.wnycMembershipAPI}/v1/emails/is-verified/?email=${email}`;
+    let headers = {'Content-Type': 'application/json'};
+    this.get('session').authorize('authorizer:nypr', (header, value) => {
+      headers[header] = value;
+    });
+    try {
+      let response = yield fetch(url, {headers, method: 'GET'});
+      if (response && response.ok) {
+        let json = yield response.json();
+        // if it's not verified, it's pending
+        this.set('emailIsPendingVerification', !get(json, 'data.is_verified'));
+      } else {
+        this.set('emailIsPendingVerification', false);
+      }
+    } catch(e) {
+      // if there's a problem with the request, return false for pending
+      // because we don't want to show the pending message and confuse users.
+      this.set('emailIsPendingVerification', false);
+    }
+  }),
+
+  emailUpdated() {
+    this.showFlash('email');
+    this.set('emailIsPendingVerification', true);
+  },
+
+  resendVerificationEmail() {
+    let url = `${config.wnycAuthAPI}/v1/confirm/resend-attr`;
+    let headers = {'Content-Type': 'application/json'};
+    this.get('session').authorize('authorizer:nypr', (header, value) => {
+      headers[header] = value;
+    });
+    return new RSVP.Promise((resolve,reject) => {
+      fetch(url, {headers, method: 'GET'}).then(response => {
+        if (response && response.ok) {
```

Comment:
It looks like you did something similar in account settings using `rejectUnsuccessfulResponses` - might be worth having something similar in web client if we find ourselves doing this more.